docs/csharp/programming-guide/statements-expressions-operators/snippets/how-to-define-value-equality-for-a-type/ValueEqualityClass/Program.cs

```diff
@@ -0,0 +1,202 @@
+﻿namespace ValueEqualityClass
+{
+    using System;
+    class TwoDPoint : IEquatable<TwoDPoint>
+    {
+        // Readonly auto-implemented properties.
+        public int X { get; private set; }
+        public int Y { get; private set; }
+
+        // Set the properties in the constructor.
+        public TwoDPoint(int x, int y)
+        {
+            if ((x < 1) || (x > 2000) || (y < 1) || (y > 2000))
```

Comment:
not very important, but this could benefit from pattern combinators. Something like:

```csharp
if (x is (< 1 or > 2000) || y is (< 1 or > 2000))
```